src/bin/server.rs

```diff
@@ -8,15 +8,59 @@
 
 use mini_redis::{server, DEFAULT_PORT};
 
+#[cfg(feature = "xray")]
+use opentelemetry::global;
+#[cfg(feature = "xray")]
+use opentelemetry::sdk::trace as sdktrace;
+#[cfg(feature = "xray")]
+use opentelemetry_aws::trace::XrayPropagator;
 use structopt::StructOpt;
 use tokio::net::TcpListener;
 use tokio::signal;
+#[cfg(feature = "xray")]
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
 
 #[tokio::main]
 pub async fn main() -> mini_redis::Result<()> {
-    // enable logging
-    // see https://docs.rs/tracing for more info
-    tracing_subscriber::fmt::try_init()?;
+    #[cfg(feature = "xray")]
+    {
```

Comment:
i would probably put all this code in a separate function, which can have two different versions based on the feature flags, rather than having all of it at the top of `main`.

Reply:
Sounds good, I'll break this out